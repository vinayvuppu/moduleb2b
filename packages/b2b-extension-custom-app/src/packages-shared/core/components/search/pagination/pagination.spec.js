import React from 'react';
import { shallow } from 'enzyme';
import RangePaginator from '../../tables/paginators/range-paginator';
import PerPageSwitcher from '../../tables/paginators/per-page-switcher';
import Pagination from './pagination';

const createTestProps = custom => ({
  totalItems: 200,
  perPage: 20,
  page: 1,
  onPerPageChange: jest.fn(),
  onPageChange: jest.fn(),
  ...custom,
});

describe('rendering', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<Pagination {...props} />);
  });

  it('outputs correct tree', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('PerPageSwitcher', () => {
    it('should receive perPage', () => {
      expect(wrapper.find(PerPageSwitcher)).toHaveProp(
        'perPage',
        props.perPage
      );
    });
    describe('calculation of itemsOnPage', () => {
      describe('when page is not the last page', () => {
        beforeEach(() => {
          props = createTestProps({ page: 1, perPage: 5, totalItems: 20 });
          wrapper = shallow(<Pagination {...props} />);
        });
        it('should equal perPage', () => {
          expect(wrapper.find(PerPageSwitcher).prop('itemsOnPage')).toBe(
            props.perPage
          );
        });
      });
      describe('when on last page', () => {
        describe('when items are left over', () => {
          beforeEach(() => {
            props = createTestProps({ page: 2, perPage: 20, totalItems: 27 });
            wrapper = shallow(<Pagination {...props} />);
          });
          it('should equal the remainder', () => {
            expect(wrapper.find(PerPageSwitcher).prop('itemsOnPage')).toBe(7);
          });
        });
      });
    });
  });

  describe('RangePaginator', () => {
    it('should receive page', () => {
      expect(wrapper.find(RangePaginator)).toHaveProp('page', props.page);
    });
  });
});
