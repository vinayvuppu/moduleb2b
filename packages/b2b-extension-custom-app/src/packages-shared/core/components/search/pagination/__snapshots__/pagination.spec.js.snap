// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering outputs correct tree 1`] = `
<div
  className="container"
>
  <div>
    <injectIntl(PerPageSwitcher)
      itemsOnPage={20}
      onPerPageChange={[MockFunction]}
      options={
        Array [
          20,
          50,
        ]
      }
      perPage={20}
    />
  </div>
  <div>
    <injectIntl(WithFormik(RangePaginator))
      onPageChange={[MockFunction]}
      page={1}
      perPage={20}
      totalItems={200}
    />
  </div>
</div>
`;
